CIP-transaction-inputs-as-list/README.md

```diff
@@ -0,0 +1,155 @@
+
+## Specification
+<!-- The technical specification should describe the proposed improvement in sufficient technical detail. In particular, it should provide enough information that an implementation can be performed solely on the basis of the design in the CIP. This is necessary to facilitate multiple, interoperable implementations. This must include how the CIP should be versioned, if not covered under an optional Versioning main heading. If a proposal defines structure of on-chain data it must include a CDDL schema in its specification.-->
+As per protocol specifications, the transaction body is structured as follows:
+```
+transaction_body =
+ { 0 : set<transaction_input>    ; inputs
+ , 1 : [* transaction_output]
+ , 2 : coin                      ; fee
+ , ? 3 : uint                    ; time to live
+ , ? 4 : [* certificate]
+ , ? 5 : withdrawals
+ , ? 6 : update
+ , ? 7 : auxiliary_data_hash
+ , ? 8 : uint                    ; validity interval start
+ , ? 9 : mint
+ , ? 11 : script_data_hash       ; New
+ , ? 13 : set<transaction_input> ; Collateral ; new
+ , ? 14 : required_signers       ; New
+ , ? 15 : network_id             ; New
+ }
+```
+
+Specifically, the inputs are currently represented as a set:
+```
+0 : set<transaction_input>    ; inputs
+```
+
+The proposed solution suggests modifying the inputs to a list format:
+```
+0 : [* transaction_input]    ; inputs
+```
+
+
+
+## Rationale: how does this CIP achieve its goals?
+<!-- The rationale fleshes out the specification by describing what motivated the design and what led to particular design decisions. It should describe alternate designs considered and related work. The rationale should provide evidence of consensus within the community and discuss significant objections or concerns raised during the discussion.
+
+It must also explain how the proposal affects the backward compatibility of existing solutions when applicable. If the proposal responds to a CPS, the 'Rationale' section should explain how it addresses the CPS, and answer any questions that the CPS poses for potential solutions.
+-->
+The motivation behind this CIP stems from the observed limitations and inefficiencies associated with the current lexicographical ordering of transaction inputs. Currently, the strict lexicographical ordering necessitates the passing of index numbers through the redeemer to select the appropriate input, leading to potential inefficiencies and vulnerabilities to vector attacks.
+
+To mitigate these issues, the proposed solution suggests transitioning from a set-based representation of transaction inputs to a list-based representation.
+
+This CIP tries to revive the original draft [CIP-0051](https://github.com/cardano-foundation/CIPs/pull/231)
```

Comment:
Do we need a new CIP to attempt to revive an old/previously not implemented CIP? What were the rationale or reasons that the earlier CIP were not adopted?

Reply:
After reviewing the referenced PR it seems like there was strong pushback against this change back in 2022 and very strong and valid arguments against this change from members of both the Plutus and Ledger teams.

Reply:
Back in 2022, hardly any projects were diving into new design patterns in order to optimize their Plutus contracts and allow for concurrent executions
But fast forward to now, most projects are adopting new design patterns that heavily rely on passing lists of indices through the redeemer, in order to identify the right inputs for their validation logic.
During 2022 the context was different, because the previous CIP didn't even have any specific use cases laid out.

Reply:
This seems like a valid rationale to at least reopen the conversation here, although I believe a lot of the criticisms about design fragility (and subsequent lack of interoperability) when building systems that are dependent on both/either input or output ordering are still valid criticisms of this approach and would need to be overcome with some overwhelming amount of evidence that this would drastically improve QoL (Quality of Life) for developers and/or users before having a hope of seeing the work necessary to accommodate this change be placed onto the roadmap.

Reply:
@solidsnakedev @Crypto2099 I'm resolving this since we have an entirely new chain of growth now... thanks to review & response... to produce a more solid candidate distinct from the original candidate https://github.com/cardano-foundation/CIPs/pull/231.